traversal/GraphTraversal.java

```diff
@@ -56,23 +54,26 @@
 import static com.vaticle.typedb.core.graph.common.Encoding.Edge.Type.SUB;
 import static com.vaticle.typeql.lang.common.TypeQLToken.Predicate.SubString.LIKE;
 
+// TODO: We should not this object as a builder, as the hash and equality functions would change
+//       We should introduce a separate "builder pattern" to Traversal, such that users of this library will build
+//       traversals with Traversal.Builder, and call .build() in the end to produce a final Object.
 public abstract class GraphTraversal extends Traversal {
 
-    final Set<Identifier.Variable.Retrievable> filter;
-
-    public GraphTraversal() {
+    GraphTraversal() {
         super();
-        filter = new HashSet<>();
     }
 
-    abstract Set<Identifier.Variable.Retrievable> filter();
+    public void filter(Modifiers.Filter filter) {
+        modifiers.filter(filter);
+    }
 
-    public abstract void filter(Set<? extends Identifier.Variable.Retrievable> filter);
+    public void sort(Modifiers.Sorting sorting) {
+        modifiers.sorting(sorting);
+    }
```

Comment:
old API `filter` takes the new type `Modifiers.Filter`, and we add an analogous `sort` API